deepsubtree.go

```diff
@@ -16,23 +17,134 @@ type DeepSubTree struct {
 	*MutableTree
 }
 
+type DSTNonExistenceProof struct {
+	*ics23.NonExistenceProof
+	LeftSiblingProof  *ics23.ExistenceProof
+	RightSiblingProof *ics23.ExistenceProof
+}
+
+// NewDeepSubTree returns a new deep subtree with the specified cache size, datastore, and version.
+func NewDeepSubTree(db dbm.DB, cacheSize int, skipFastStorageUpgrade bool, version int64) (*DeepSubTree, error) {
```

Comment:
Is this satisfying an interface? If not, I would drop the return error as it isn't used.